types/dom-background-sync/index.d.ts

```diff
@@ -0,0 +1,31 @@
+// Type definitions for non-npm package dom-background-sync-browser 0.0
+// Project: https://wicg.github.io/background-sync/spec/#api-description
+// Definitions by: Hana Joo <https://github.com/h-joo>
+//                 Jan Kuehle <https://github.com/frigus02>
+//                 Martin Probst <https://github.com/mprobst>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.4
+
+interface SyncEventInit extends ExtendableEventInit {
+    lastChance?: boolean;
+    tag: string;
+}
+
+// https://wicg.github.io/background-sync/spec/#syncevent
+interface SyncEvent extends ExtendableEvent {
+    readonly lastChance: boolean;
+    readonly tag: string;
+}
+
+declare var SyncEvent: { prototype: SyncEvent; new (type: string, init?: SyncEventInit): SyncEvent };
+
+// https://wicg.github.io/background-sync/spec/#syncmanager
+interface SyncManager {
```

Comment:
I think we may also want `declare var SyncManager`. Similar to how it was defined in TypeScript 4.3: https://github.com/microsoft/TypeScript/blob/1294ee8c1f37c8bf214db67e180edb3220a52236/lib/lib.webworker.d.ts#L3069-L3077

This seems like a super minor use case because `new SyncManager()` throws an exception. But the constructor does exist and `SyncManager.prototype` exists, too.